### PR TITLE
added albums service, added & refactored errors structure

### PR DESCRIPTION
**Pull request by andreswolox**
## Summary  [Card](https://trello.com/c/jiPu6c0V/38-integraci%C3%B3n-con-api-de-albumes-servicios-promises)

- Added `Albums` Service
- Added `GetAlbums` function for Albums Service
- Added `GetPhotos` function for Albums Service
- Created `errors` folder
- Refactored `errors.js` (moved to index of `errors` folder)
- Destructured existing errors by type
- added `AxiosError`

### CODE REVIEW

File: app/helpers/request.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const axios = require('axios');
+const { errors } = require('../errors');
+
+const buildRequest = ({ url, path = '', method, headers, data, params }) => ({
```
Comment (jersonlopez) [inline app/helpers/request.js]: Do you remember what we spoke about at the meeting?

File: app/helpers/request.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  params
+});
+
+module.exports = options =>
+  axios(buildRequest(options))
+    .then(info => info.data)
+    .catch(error => {
+      const { statusText } = error.response;
+      const message = error.response.data.error;
+      throw errors.axiosError(message, statusText);
```
Comment (jersonlopez) [inline app/helpers/request.js]: The second param is unnecessary, check this file `app/errors/axios.js` please.

File: app/services/albums.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const request = require('../helpers/request');
+const logger = require('../logger');
+
+const JPH_API_PATH = process.env.JSON_PLACE_HOLDER_API_PATH;
```
Comment (jersonlopez) [inline app/services/albums.js]: Please check this folder: `al-express-js/config/`, there is a way to handle all environment variables

File: app/middlewares/errors.js [hunk 4]
```diff
@@ -5,7 +5,8 @@ const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
-  [errors.DEFAULT_ERROR]: 500
+  [errors.DEFAULT_ERROR]: 500,
+  [errors.AXIOS_ERROR]: 800
```
Comment (jersonlopez) [inline app/middlewares/errors.js]: what does mean 800 status code? I think that is better 503, what do you think?

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: app/helpers/request.js (hunk 1 above)
Comment (andreswolox) [reply app/helpers/request.js]: Of course, actually, I've changed param names and already commited in future next card in another branch, but no problem, already changed here too! :100: 

- Changed `url` param name to `baseUrl` for better understanding. 
- Kept `path` param the same name based on Swagger Doc I sent to you.

If you have more observations or you are not agree, I'd appreciate letting me know it.

File: app/helpers/request.js (hunk 2 above)
Comment (andreswolox) [reply app/helpers/request.js]: Sorry for hear that, already fixed it! My Stupid 'layer 8'.

File: app/services/albums.js (hunk 3 above)
Comment (andreswolox) [reply app/services/albums.js]: Another 'layer 8'. How could I've forgotten, sorry for that.

Comment (andreswolox) [reply app/services/albums.js]: Fixed it!

File: app/middlewares/errors.js (hunk 4 above)
Comment (andreswolox) [reply app/middlewares/errors.js]: Hahaha actually, I was not sure what statusCode set. 
Thanks so much for the feedback! Fixed it!

File: config/index.js [hunk 5]
```diff
@@ -45,6 +45,9 @@ const config = {
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
     }
+  },
+  jsonPlaceHolderApi: {
+    baseUrl: process.env.JSON_PLACE_HOLDER_API_BASE_URL
```
Comment (jersonlopez) [inline config/index.js]: what happens if you have a different API URL for dev and master?

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: config/index.js (hunk 5 above)
Comment (andreswolox) [reply config/index.js]: Fixed it! :)

**Review by jersonlopez — APPROVED** [review]
good job bro 🥇 💯

**Pull request merged**